libs/server/Lua/LuaRunner.cs

```diff
@@ -0,0 +1,396 @@
+                        _ = api.SET(key, value);
+                        return "OK";
+                    }
+                case "GET":
+                case "get":
+                    {
+                        if (!respServerSession.CheckACLPermissions(RespCommand.GET))
+                            throw new Exception(Encoding.ASCII.GetString(CmdStrings.RESP_ERR_NOAUTH));
+                        var key = scratchBufferManager.CreateArgSlice(Convert.ToString(args[0]));
+                        var status = api.GET(key, out var value);
+                        if (status == GarnetStatus.OK)
+                            return value.ToString();
+                        return null;
+                    }
+                // As fallback, we use RespServerSession with a RESP-formatted input. This could be optimized
+                // in future to provide parse state directly.
+                default:
+                    {
+                        var request = scratchBufferManager.FormatCommandAsResp(cmd, args, state);
+                        _ = respServerSession.TryConsumeMessages(request.ptr, request.length);
+                        var response = scratchBufferNetworkSender.GetResponse();
+                        var result = ProcessResponse(response.ptr, response.length);
+                        scratchBufferNetworkSender.Reset();
+                        return result;
+                    }
+            }
+        }
+
+        /// <summary>
+        /// Process a RESP-formatted response from the RespServerSession
+        /// </summary>
+        unsafe object ProcessResponse(byte* ptr, int length)
+        {
+            switch (*ptr)
+            {
+                case (byte)'+':
+                    if (RespReadUtils.ReadSimpleString(out var resultStr, ref ptr, ptr + length))
+                        return resultStr;
+                    break;
+                case (byte)':':
+                    if (RespReadUtils.ReadIntegerAsString(out var resultInt, ref ptr, ptr + length))
+                        return resultInt;
+                    break;
+                case (byte)'-':
+                    if (RespReadUtils.ReadErrorAsString(out resultStr, ref ptr, ptr + length))
+                        return resultStr;
+                    break;
+
+                case (byte)'$':
+                    if (RespReadUtils.ReadStringResponseWithLengthHeader(out resultStr, ref ptr, ptr + length))
+                        return resultStr;
+                    break;
+
+                case (byte)'*':
+                    if (RespReadUtils.ReadStringArrayResponseWithLengthHeader(out var resultArray, ref ptr, ptr + length))
+                        return resultArray;
+                    break;
+
+                default:
+                    throw new Exception("Unexpected response: " + Encoding.UTF8.GetString(new Span<byte>(ptr, length)).Replace("\n", "|").Replace("\r", "") + "]");
```

Comment:
consider introducing a regex for `\n\r` and then calling `regex.Replace`

Reply:
This could be done across the codebase. Leaving for future optimization as it is a non-critical code path.